src/stp/new_plays/FreeKickThem.cpp

```diff
@@ -0,0 +1,129 @@
+//
+// Created by jordi on 07-05-20.
+//
+
+#include "stp/new_plays/FreeKickThem.h"
+
+#include "stp/invariants/game_states/FreeKickThemGameStateInvariant.h"
+#include "stp/new_roles/Keeper.h"
+#include "stp/new_roles/Defender.h"
+#include "stp/new_roles/Formation.h"
+
+namespace rtt::ai::stp::play {
+
+FreeKickThem::FreeKickThem() : Play() {
+    startPlayInvariants.clear();
+    startPlayInvariants.emplace_back(std::make_unique<invariant::FreeKickThemGameStateInvariant>());
+
+    keepPlayInvariants.clear();
+    keepPlayInvariants.emplace_back(std::make_unique<invariant::FreeKickThemGameStateInvariant>());
+
+    roles = std::array<std::unique_ptr<Role>, rtt::ai::Constants::ROBOT_COUNT()>{
+            std::make_unique<role::Keeper>(role::Keeper("keeper")),
+            std::make_unique<role::Defender>(role::Defender("defender_0")),
+            std::make_unique<role::Defender>(role::Defender("defender_1")),
+            std::make_unique<role::Defender>(role::Defender("defender_2")),
+            std::make_unique<role::Defender>(role::Defender("blocker_0")),
+            std::make_unique<role::Defender>(role::Defender("blocker_1")),
+            std::make_unique<role::Defender>(role::Defender("blocker_2")),
+            std::make_unique<role::Defender>(role::Defender("blocker_3")),
+            std::make_unique<role::Defender>(role::Defender("blocker_4")),
+            std::make_unique<role::Defender>(role::Defender("blocker_5")),
+            std::make_unique<role::Formation>(role::Formation("offender"))};
+}
+
+uint8_t FreeKickThem::score(world_new::World* world) noexcept { return 100; }
+
+void FreeKickThem::calculateInfoForRoles() noexcept {
+    calculateInfoForKeeper();
+    calculateInfoForBlockers();
+    calculateInfoForDefenders();
+    calculateInfoForOffenders();
+}
+
+void FreeKickThem::calculateInfoForKeeper() noexcept {
+    stpInfos["keeper"].setEnemyRobot(world->getWorld()->getRobotClosestToBall(world_new::them));
+    stpInfos["keeper"].setPositionToShootAt(Vector2());
+}
+
+void FreeKickThem::calculateInfoForBlockers() noexcept {
+    const double MINIMAL_DISTANCE_FROM_BALL = 0.5;
```

Comment:
What is the measure used for this variable? Feet, cm, cows, ...?

Reply:
I think cows

Reply:
It could be good to make some standard for distances. I always assume they are in meters, because that makes the most sense. If that is always the case, we can add this to the readme for AI. I know in Roboteam_World many things are in cm or mm suddenly. But this is not necessary to do before the 10th of June so I'll make a long-term issue out of it